sympy/utilities/codegen.py

```diff
@@ -859,8 +902,53 @@ def _declare_globals(self, routine):
         return []
 
     def _declare_locals(self, routine):
-        # loop variables are declared in loop statement
-        return []
+
+        # Compose a list of symbols to be dereferenced in the function
+        # body. These are the arguments that were passed by a reference
+        # pointer, excluding arrays.
+        dereference = []
+        for arg in routine.arguments:
+            if isinstance(arg, ResultBase) and not arg.dimensions:
+                dereference.append(arg.name)
+        
+        code_lines = []
+        for result in routine.local_vars:
+
+            if isinstance(result, Result):
+                assign_to = result.name
+                t = result.get_datatype('c')
+                if isinstance(result.expr, (MatrixBase, MatrixExpr)):
+                    dims = result.expr.shape
+                    assert dims[1] == 1, "2D matrices not supported yet"
```

Comment:
change not to use assert and maybe "only column vectors supported for now" (since matrices are 2D by definition)